internal/hcapi2/mock/mock_gen.go

```diff
@@ -1,5 +1,6 @@
 package hcapi2_mock
 
+//go:generate mockgen -package hcapi2_mock -destination zz_config_mock.go github.com/hetznercloud/cli/internal/state/config Config
```

Comment:
I dont think this belongs into `internal/hcapi2/mock`. Its not a mock for `hcapi`. Perhaps `internal/state/config/mock`?

Reply:
Yes, I already realized that :D